doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,108 @@
+
+* `ARM TrustZone`_ technology and hardware accelerated firmware isolation
+*  :ref:`trusted_storage_readme` |NCS| software library
+
+Currently all :ref:`Matter samples <matter_samples>` use Trusted Storage library as a trusted storage backend for all supported platforms.
+
+Trusted Storage library provides confidentiality, integrity, and authenticity for the assets it operates on.
+It handles sensitive data in two contexts:
+
+* volatile - before data is forwarded to the non-volatile storage and after it is retrieved from the non-volatile storage
+* non-volatile - while data is being written to the non-volatile memory
+
+In the case of volatile context, the Trusted Storage library leverages the Authenticated Encryption with Associated Data (AEAD) encryption backend (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD`).
+AEAD can be configured with the set of Kconfig options described in the :ref:`Trusted Storage configuration <trusted_storage_configuration>` section.
+
+An important setting, that depends on the hardware platform in use, is the way of generating the AEAD keys.
+The recommended and the most secure option is to use :ref:`Hardware Unique Key (HUK) library <lib_hw_unique_key>` for that purpose.
+HUK support is automatically enabled with :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible platforms.
+Note that, due to hardware limitations, this library is not available for nRF52840 family.
+In that case, the AEAD keys can be derived with SHA-256 hash (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`) but this approach is less secure than using HUK for key derivation as it will only provide integrity of sensitive material.
+It is also possible to implement a custom AEAD keys generation method when the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_CUSTOM` Kconfig option is selected.
+
+You can find an overview of Trusted Storage layer configuration supported for each |NCS| Matter-enabled platform in `Matter platforms security support`_ section.
+
+Matter platforms security support
+*********************************
+
+The following table summarizes current security configuration and features supported for Matter-enabled hardware platforms in |NCS|.
+
+.. list-table:: Matter platforms security support
+   :widths: auto
+   :header-rows: 1
+
+   * - Platform
+     - Networking backend
+     - Cryptography API
+     - Cryptography backend
+     - ARM TrustZone support
+     - Trusted Storage backend
+     - Trusted Storage AEAD key
+   * - nRF52840
+     - Thread
+     - PSA
+     - Oberon
+     - No
+     - Trusted Storage library
+     - SHA-256 hash
+   * - nRF5340
+     - Thread
+     - PSA
+     - Oberon
+     - Yes
+     - Trusted Storage library
+     - Hardware Unique Key (HUK)
+   * - nRF5340 + nRF7002
+     - Wi-Fi
+     - Mbed TLS
+     - Oberon
+     - Yes
+     - Trusted storage library
+     - Hardware Unique Key (HUK)
```

Comment:
Are you sure that we use HUK for the WiFi variant? I guess it may be correct regarding internal WiFi security, but I'm not sure that all of them use HUK.

Reply:
Trusted Storage is not used for our Wi-Fi solution yet. We also discussed few other points about HUK which needs to be changed F2F, so i believe one more update is comming :)